lib/rubygems/config_file.rb

```diff
@@ -276,7 +276,12 @@ def check_credentials_permissions
   # Location of RubyGems.org credentials
 
   def credentials_path
-    File.join Gem.user_home, '.gem', 'credentials'
+    credentials = File.join Gem.user_home, '.gem', 'credentials'
+    if File.exist? credentials
+      credentials
+    else
+      File.join Gem.config_home, "gem", "credentials"
```

Comment:
I don't think that $XDG_CONFIG_HOME is the proper location for the credentials file. From what I understand, the $XDG_CONFIG_HOME folder should only contain application settings and configuration, and not something like a user's API keys. The credential file stores API Keys, which means it should be in $XDG_DATA_HOME

Reply:
:+1: I try to update to use the `XDG_CACHE_HOME` for storing the credentials.